modules/globebrowsing/rendering/layer/layer.cpp

```diff
@@ -34,56 +34,285 @@ namespace {
     const char* keyEnabled = "Enabled";
     const char* keyLayerGroupID = "LayerGroupID";
     const char* keySettings = "Settings";
+    const char* keyAdjustment = "Adjustment";
+    const char* KeyBlendMode = "BlendMode";
 }
 
-Layer::Layer(layergroupid::ID id, const ghoul::Dictionary& layerDict)
+Layer::Layer(layergroupid::GroupID id, const ghoul::Dictionary& layerDict)
     : properties::PropertyOwner(layerDict.value<std::string>(keyName))
+    , _typeOption(
+        "type",
+        "Type",
+        properties::OptionProperty::DisplayType::Dropdown
+    )
+    , _blendModeOption(
+        "blendMode",
+        "Blend Mode",
+        properties::OptionProperty::DisplayType::Dropdown
+    )
     , _enabled(properties::BoolProperty("enabled", "Enabled", false))
     , _reset("reset", "Reset")
+    , _tileProvider(nullptr)
+    , _otherTypesProperties{
+        properties::Vec3Property (
+        "color",
+        "Color",
+        glm::vec4(1.f, 1.f, 1.f, 1.f),
+        glm::vec4(0.f),
+        glm::vec4(1.f))
+    }
+    , _layerGroupId(id)
+    , _onChangeCallback([](){})
```

Comment:
You can check:
```
if (_onChangeCallback) {
   _onChangeCallback();
}
```

If the callback has not been set, it evaluates to `false`.  An empty lambda expression evaluates to `true`